### PR TITLE
Change repl to use a fixup_bucket function

### PR DESCRIPTION
**Pull request by Vagabond**

### CODE REVIEW

File: src/riak_repl.erl [hunk 1]
```diff
@@ -18,7 +19,38 @@ install_hook() ->
...
+strip_postcommit(BucketProps) ->
+    %% Get the current postcommit hook
+    case proplists:get_value(postcommit, BucketProps, []) of
+        X when is_list(X) ->
+            CurrentPostcommit=X;
+        {struct, _}=X ->
+            CurrentPostcommit=[X]
+    end,
     
+    %% Add kv/search hook - make sure there are not duplicate entries
```
Comment (rzezeski) [inline src/riak_repl.erl]: What's up with the search reference?

**Comment by rzezeski** [comment]
+1 to merge

File: src/riak_repl.erl [hunk 2]
```diff
@@ -18,7 +19,38 @@ install_hook() ->
...
-                        proplists:delete(postcommit, DefaultBucketProps)),
-    riak_core_bucket:append_bucket_defaults([{postcommit, [?REPL_HOOK]}]),
+                        proplists:delete(repl, DefaultBucketProps)),
+    riak_core_bucket:append_bucket_defaults([{repl, true}]),
     ok.
+
+fixup(_Bucket, BucketProps) ->
+    case proplists:get_value(repl, BucketProps) of
+        true ->
+            CleanPostcommit = strip_postcommit(BucketProps),
```
Comment (jonmeredith) [inline src/riak_repl.erl]: Clean/Update portions are common to both cases - but duplication isn't huge.

**Comment by jonmeredith** [comment]
Looks good to me - fiddled around with the HTTP interface to get/set repl properties.
+1 merge after I've checked the core/search stuff.

**Pull request merged**